src/PostgREST/ApiRequest.hs

```diff
@@ -122,6 +122,10 @@ userApiRequest schema req reqBody =
               Nothing -> PayloadParseError "All lines must have same number of fields"
               Just json -> PayloadJSON json)
             (CSV.decodeByName reqBody)
+        Left "application/x-www-form-urlencoded" ->
```

Comment:
That's an interesting point and one I wrestled with. The issue is that `ContentType` is used not only for code involving the request Content-Type header as this PR is, but also code that checks the Accept header and chooses the *response* Content-Type header. Extending `ContentType` with a `FormUrlEncoded` constructor would mean I have to add a case in the show instance for it like

```haskell
data ContentType = ApplicationJSON | TextCSV | FormUrlEncoded deriving Eq
instance Show ContentType where
  show ApplicationJSON = "application/json; charset=utf-8"
  show TextCSV         = "text/csv; charset=utf-8"
  show FormUrlEncoded  = "application/x-www-form-urlencoded; charset=utf-8"
```

but this seems misleading, like we may be planning to use it in responses or something. It is possible to do this and leave the `unsupportedAccept` middleware as it is now which would prevent people from choosing form urlencoding in an Accept header. However the code communicates the wrong intention. Either way seems awkward and I think I got myself into a corner with the types as I have defined them.